arangod/Graph/Providers/ClusterProvider.cpp

```diff
@@ -262,15 +264,31 @@ void ClusterProvider<StepImpl>::destroyEngines() {
 }
 
 template<class StepImpl>
-Result ClusterProvider<StepImpl>::fetchEdgesFromEngines(
-    VertexType const& vertex) {
+Result ClusterProvider<StepImpl>::fetchEdgesFromEngines(Step* step) {
+  TRI_ASSERT(step != nullptr);
+
   auto const* engines = _opts.engines();
-  // TODO Assert that the vertex is not in _vertexConnections after no-loose-end
+  // TODO Assert that the step is not in _vertexConnections after no-loose-end
   // handling todo is done.
   transaction::BuilderLeaser leased(trx());
   leased->openObject(true);
-  leased->add("backward", VPackValue(_opts.isBackward()));
-  leased->add("keys", VPackValue(vertex.toString()));
+  leased->add("backward",
+              VPackValue(_opts.isBackward()));  // [GraphRefactor] ksp only?
```

Comment:
What is the question? `ksp only?`

Reply:
I think it means: "In use of KShortestPath only?" -> And the answer is partially yes. All TwoSidedEnumerator related will potentially need it.

Reply:
Ah yes, that is because the Edges API distinguishes this, okay I remember